packages/nodes-base/nodes/Slack/V2/SlackV2.node.ts

```diff
@@ -1065,23 +1070,67 @@ export class SlackV2 implements INodeType {
 							} else {
 								uploadData = Buffer.from(binaryData.data, BINARY_ENCODING);
 							}
-							body.file = {
-								value: uploadData,
-								options: {
-									filename: binaryData.fileName,
-									contentType: binaryData.mimeType,
-								},
-							};
-							responseData = await slackApiRequest.call(
-								this,
-								'POST',
-								'/files.upload',
-								{},
-								qs,
-								{ 'Content-Type': 'multipart/form-data' },
-								{ formData: body },
-							);
-							responseData = responseData.file;
+
+							if (nodeVersion <= 2.1) {
+								body.file = {
+									value: uploadData,
+									options: {
+										filename: binaryData.fileName,
+										contentType: binaryData.mimeType,
+									},
+								};
+
+								responseData = await slackApiRequest.call(
+									this,
+									'POST',
+									'/files.upload',
+									{},
+									qs,
+									{ 'Content-Type': 'multipart/form-data' },
+									{ formData: body },
+								);
+								responseData = responseData.file;
+							} else {
+								fileBody.file = {
+									value: uploadData,
+									options: {
+										filename: binaryData.fileName,
+										contentType: binaryData.mimeType,
+									},
+								};
+								const uploadUrl = await slackApiRequest.call(
+									this,
+									'GET',
+									'/files.getUploadURLExternal',
+									{},
+									{
+										filename: options.fileName ? options.fileName : binaryData.fileName,
+										length: binaryData.fileSizeRaw,
```

Comment:
I'd prefer that we remove `fileSizeRaw`, and use `getBinaryMetadata` like we do in the [Google Drive Node](https://github.com/n8n-io/n8n/blob/master/packages/nodes-base/nodes/Google/Drive/v1/GoogleDriveV1.node.ts#L2445-L2446) .
We should eventually fix this issue of `fileSize` being a string instead of the actual number of bytes, but that should happen for all of the code. Adding this additional field is creating new tech-debt.